elpy.el

```diff
@@ -1737,6 +1737,84 @@ If ASK-FOR-EACH-ONE is non-nil, ask before killing each python process.
+  "Profile asynchronously the active region or the current buffer
+and display the result using `elpy-profile-visualizer'.
+
+If IN-DIR is non nil, profile result is saved in the same
+directory as the script.
+If DONT-DISPLAY is non nil, don't display the profile results."
+  (interactive "P")
+  (let* ((file-name (buffer-name))
+         (file-dir (file-name-directory (buffer-file-name)))
+	 (beg (if (region-active-p) (region-beginning) (point-min)))
+	 (end (if (region-active-p) (region-end) (point-max)))
+	 (tmp-file-prefix (if (region-active-p) "_region_" ""))
+	 (tmp-file (if in-dir (concat file-dir "/" tmp-file-prefix file-name)
+                     (concat (make-temp-file "elpy-profile-" t)  "/" tmp-file-prefix file-name)))
+         (region (elpy-shell--region-without-indentation beg end)))
+    (with-temp-buffer
+      (insert region)
+      (write-region (point-min) (point-max) tmp-file nil t))
+    (elpy-profile--file tmp-file t dont-display) dont-display))
+
+(defun elpy-profile--file (file &optional in-dir dont-display)
+  "Profile asynchronously FILE and display the result using
+`elpy-profile-visualizer'.
+
+If IN-DIR is non nil, profile result is saved in the same
+directory as the script.
+If DONT-DISPLAY is non nil, don't display the profile results."
+  ;; clean log buffers
+  (condition-case nil
+      (with-current-buffer " *elpy-profile-log*"
+	(kill-this-buffer))
+    (error nil))
+  (condition-case nil
+      (with-current-buffer "*elpy-profile-log*"
+	(kill-this-buffer))
+    (error nil))
+  (let* ((prof-file (if in-dir (concat (file-name-sans-extension file) ".profile")
+		      (concat (make-temp-file "elpy-profile-" nil ".profile"))))
+	 (proc-name (format "elpy-profile-%s" file))
+	 (proc-cmd (list python-shell-interpreter "-m" "cProfile" "-o" prof-file file))
+	 (proc (make-process :name proc-name
+			     :buffer " *elpy-profile-log*"
+			     :sentinel 'elpy-profile--sentinel
+			     :command proc-cmd)))
+    (message "[%s] Profiling ..." (file-name-nondirectory file))
+    (process-put proc 'prof-file prof-file)
+    (process-put proc 'file file)
+    (process-put proc 'dont-display dont-display)))
+
+(defun elpy-profile--display-profiling (file)
+  "Display the profile result FILE using `elpy-profile-visualizer'."
+  (if (executable-find elpy-profile-visualizer)
+      (call-process elpy-profile-visualizer nil 0 nil file)
+    (message "Elpy profile visualizer '%s' not found" elpy-profile-visualizer)))
+
+(defun elpy-profile--sentinel (process string)
+  "Elpy profile sentinel."
+  (let ((filename (file-name-nondirectory (process-get process 'file)))
+	(prof-file (process-get process 'prof-file))
+        (dont-display (process-get process 'dont-display)))
+    (if (not (string-equal string "finished\n"))
+	(progn
+	  (message  "[%s] Profiling failed" filename)
+	  (switch-to-buffer-other-window " *elpy-profile-log*")
+	  (read-only-mode)
+	  (rename-buffer "*elpy-profile-log*"))
```

Comment:
I let the buffer visible as suggested, and replace `read-only-mode` by `view-mode`